wallet/src/main/java/com/android/identity_credential/wallet/BiometricUserAuthPrompt.kt

```diff
@@ -0,0 +1,149 @@
+    private var onCanceled: () -> Unit = {}
+    private var onError: (Throwable) -> Unit = { exception -> throw exception }
+
+    private var lskfOnNegativeBtn: Boolean = false
+
+    // stored in order to be reused again in case of lskf fallback
+    private lateinit var biometricPrompt: BiometricPrompt
+    private var cryptoObject: BiometricPrompt.CryptoObject? = null
+    private lateinit var title: String
+
+    private var biometricAuthCallback = object : BiometricPrompt.AuthenticationCallback() {
+        override fun onAuthenticationError(
+            errorCode: Int,
+            errorString: CharSequence
+        ) {
+            super.onAuthenticationError(errorCode, errorString)
+            if (errorCode == BiometricPrompt.ERROR_NEGATIVE_BUTTON && lskfOnNegativeBtn) {
+                // if no delay is injected, then biometric prompt's auth callbacks would not be called
+                Handler(Looper.getMainLooper()).postDelayed({
+                    authenticateLskf()
+                }, 100)
+            } else if (errorCode == BiometricPrompt.ERROR_USER_CANCELED) {
+                onCanceled.invoke()
+            } else {
+                onError.invoke(IllegalStateException(errorString.toString()))
+            }
+        }
+
+        override fun onAuthenticationSucceeded(
+            result: BiometricPrompt.AuthenticationResult
+        ) {
+            super.onAuthenticationSucceeded(result)
+            onSuccess()
+        }
+
+        override fun onAuthenticationFailed() {
+            super.onAuthenticationFailed()
+            onCanceled.invoke()
+        }
+    }
+
+    /**
+     * Prompts user for authentication, and calls the provided functions when authentication is
+     * complete. Biometric authentication will be offered first if both [UserAuthenticationType.LSKF]
+     * and [UserAuthenticationType.BIOMETRIC] are allowed.
+     *
+     * @param activity the activity hosting the authentication prompt
+     * @param title the title for the authentication prompt
+     * @param cryptoObject a crypto object to be associated with this authentication
+     * @param userAuthenticationTypes the set of allowed user authentication types, must contain at
+     *                                least one element
+     * @param onSuccess the function which will be called when the user successfully authenticates
+     * @param onCanceled the function which will be called when the user cancels
+     * @param onError the function which will be called when there is an unexpected error in the user
+     *                authentication process - a throwable will be passed into this function
+     */
+    fun authenticate(
+        activity: FragmentActivity,
+        title: String,
+        cryptoObject: BiometricPrompt.CryptoObject?,
+        userAuthenticationTypes: Set<UserAuthenticationType>,
+        onSuccess: () -> Unit,
+        onCanceled: () -> Unit,
+        onError: (Throwable) -> Unit,
+    ) {
```

Comment:
So in [this comment](https://github.com/openwallet-foundation-labs/identity-credential/pull/512#discussion_r1516753726) I suggested that we should just have a single function `showBiometricPrompt()` and you seemed to agree but it still looks like this

```
val prompt = BiometricIUserAuthPrompt()
prompt.authenticate(XYZ)
```

Looks like you are almost there and you just need to make `BiometricUserAuthPrompt` private and create a public function `showBiometricPrompt(XYZ)` which internally does the above...

Second, we need to take a `val requireConfirmation: Boolean` (and pass `false` in our app) and this should be passed to [setConfirmationRequired() on BiometricPrompt.PromptInfo.Builder](https://developer.android.com/reference/androidx/biometric/BiometricPrompt.PromptInfo.Builder#setConfirmationRequired(boolean)) ... this is useful with Face Auth on e.g. Pixel 8 and later.